docs/model_repository.md

```diff
@@ -169,6 +171,68 @@ $ export AZURE_STORAGE_ACCOUNT="account_name"
 $ export AZURE_STORAGE_KEY=$(az storage account keys list -n $AZURE_STORAGE_ACCOUNT --query "[0].value")
 ```
 
+### Cloud Storage with Credential file (Beta)
+
+*This feature is currently in beta and may be subject to change.*
+
+To group the credentials into a single file for Triton, you may set the 
+`TRITON_CLOUD_CREDENTIAL_PATH` environment variable to a path pointing to a 
+JSON file of the following format, residing in the local file system.
+
+```
+export TRITON_CLOUD_CREDENTIAL_PATH="cloud_credential.json"
+```
+
+"cloud_credential.json":
+```
+{
+  "gs": {
+    "": "PATH_TO_GOOGLE_APPLICATION_CREDENTIALS",
+    "gs://gcs-bucket-002": "PATH_TO_GOOGLE_APPLICATION_CREDENTIALS_2"
+  },
+  "s3": {
+    "": {
+      "secret_key": "AWS_SECRET_ACCESS_KEY",
+      "key_id": "AWS_ACCESS_KEY_ID",
+      "region": "AWS_DEFAULT_REGION",
+      "session_token": "",
+      "session_token": "",
+    },
+    "s3://s3-bucket-002": {
+      "secret_key": "AWS_SECRET_ACCESS_KEY_2",
+      "key_id": "AWS_ACCESS_KEY_ID_2",
+      "region": "AWS_DEFAULT_REGION_2",
+      "session_token": "AWS_SESSION_TOKEN_2"
+      "session_token": "AWS_PROFILE_2"
+    }
+  },
+  "as": {
+    "": {
+      "account_str": "AZURE_STORAGE_ACCOUNT",
+      "account_key": "AZURE_STORAGE_KEY"
+    },
+    "as://Account-002/Container": {
+      "account_str": "",
+      "account_key": ""
+    }
+  }
+}
+```
+
+To match a credential, the longest matching credential name against the start
+of a given path is used. For example: `gs://gcs-bucket-002/model_repository`
+will match the "gs://gcs-bucket-002" GCS credential, and
+`gs://any-other-gcs-bucket` will match the "" GCS credential.
```

Comment:
Is "" a required field in the file if the user specifies to use the credential file?

Reply:
+1

Reply:
It is not required, but "" is good for wildcard matching.

Nothing in the credential file is required. If the loading cannot find an "expected" item(s), it is left empty and let the client wrapper constructor to figure out what to do. It is similar to missing an environment variable in the current setup.